glide-core/redis-rs/redis/src/commands/cluster_scan.rs

```diff
@@ -130,11 +146,18 @@ impl ScanStateRC {
     }
 }
 
-/// This trait defines the methods for interacting with a Redis cluster during scanning.
-#[async_trait]
+/// Trait defining methods for interacting with a cluster during scanning.
+///
+/// Implementors of this trait provide functionalities such as retrieving the address
+/// for a slot, getting the epoch of an address, routing commands, and checking cluster coverage.
 pub(crate) trait ClusterInScan {
     /// Retrieves the address associated with a given slot in the cluster.
-    async fn get_address_by_slot(&self, slot: u16) -> RedisResult<Arc<String>>;
+    async fn get_address_by_slot(
+        &self,
+        slot: u16,
+        scanned_slots_map: &mut SlotsBitsArray,
+        allow_non_covered_slots: bool,
+    ) -> RedisResult<Arc<String>>;
```

Comment:
Question: does `Arc` required here, or can we use `Rc` ? IIRC, `String` is not thread safe nor it implements the `Copy` trait - so we are not using this method among threads. If we are using `Arc` for ref counting, then `Rc` is more than enough and it is more efficient

Reply:
It's a wider change, this Arc is used everywhere in the core for adrrs strings or slots.
I can change here, but ill need to make it arc back in the internal core functions.